UI/src/app/services/backend-connector.service.ts

```diff
@@ -63,20 +59,126 @@ export class BackendConnectorService {
         this._layout.next(map);
     }
 
-    public requestData(userInput: string, datasetId: number){
-        return this.http
-            .post('localhost:3001/api', JSON.stringify({userinput: userInput, datasetid: datasetId}))
-            .retry(this.connectionRetries)
-            .timeout(this.connectionTimeout)
-            .map((response: Response) =>
-                response.json()
-            )
+    public requestData(userInput: string, datasetName: string, that){
+        return new Promise(function(reject, fullfill){
+            this.that = that;
+            this.http
+            .get('http://localhost:3001/api/nlptodata', {
+                params: {
+                    userquery: userInput,
+                     dataset: datasetName
+                }
+            })
             .subscribe((data) => {
-                console.log(data);
+                console.log(this.that);
+                var t:Trace = new Trace(data["columnA"], data["columnB"], data["plotType"])
+                this.parseData(t)
```

Comment:
oh I see, I divided send and request, but we should do it as one, thats correct, maybe we should add the deleted lines from the getData methode in here